packages/errors/src/errors/RequestError.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  parseErrorResponse,
+  ErrorResponse,
+  deserializeError,
+} from '../serialization';
+
+/**
+ * An error thrown as the result of a failed server request.
+ *
+ * The server is expected to respond on the ErrorResponse format.
+ */
+export class RequestError extends Error {
```

Comment:
Hmm, is a request error not an error in sending the request, like not being able to resolve a DNS name? Thinking ResponseError works better

Reply:
Haha, sure, can rename

Reply:
done